lab_4_retrieval_w_clustering/main.py

```diff
@@ -182,6 +260,19 @@ def retrieve_relevant_documents(self, query: str, n_neighbours: int) -> list[tup
         Returns:
             list[tuple[float, str]]: Relevant documents with their distances.
         """
+        if not query or not isinstance(query, str) or not n_neighbours\
+                or not isinstance(n_neighbours,int) or n_neighbours <= 0:
+            raise ValueError
+        tokenized_query = self._db.get_tokenizer().tokenize(query)
```

Comment:
use information that we are inheriting from the `BasicSearchEngine` class

Reply:
wow, true